File: android/src/main/java/com/lib/flutter_blue_plus/l2cap/messages/OpenL2CapChannelRequest.java
// Copyright 2024, Continental Automotive Technologies GmbH
// All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
package com.lib.flutter_blue_plus.l2cap.messages;

import com.lib.flutter_blue_plus.l2cap.L2CapAttributeNames;

import java.util.Map;

public class OpenL2CapChannelRequest {
    public final String remoteId;
    public final int psm;
    public final boolean secure;

    public OpenL2CapChannelRequest(String remoteId, int psm, boolean secure) {
        this.remoteId = remoteId;
        this.psm = psm;
        this.secure = secure;
    }

    public static OpenL2CapChannelRequest unmarshal(final Map<String, Object> data) {
        final String remoteId = (String) data.get(L2CapAttributeNames.KEY_REMOTE_ID);
        final int psm = (int) data.get(L2CapAttributeNames.KEY_PSM);
        final boolean secure = (boolean) data.get(L2CapAttributeNames.KEY_SECURE);
        return new OpenL2CapChannelRequest(remoteId, psm, secure);
    }


}

File: ios/Classes/l2capmessages/OpenL2CapChannelRequest.swift
// Copyright 2023, Continental Automotive Technologies GmbH
// All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

import Foundation

@objc
public class OpenL2CapChannelRequest: NSObject {
    
    @objc
    public let secure: Bool
    @objc
    public let remoteId: String
    @objc
    public let psm: Int;
    
    @objc
    public init (data: NSDictionary) {
        self.secure = data[L2CapAttributeNames.keySecure] as? Bool ?? false
        self.psm = data[L2CapAttributeNames.keyPsm] as? Int ?? 0
        self.remoteId = data[L2CapAttributeNames.keyRemoteId] as? String ?? ""
    }
                
}
